pkg/tracing/otel_trace_provider.go

```diff
@@ -0,0 +1,76 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/justtrackio/gosoline/pkg/appctx"
+	"github.com/justtrackio/gosoline/pkg/cfg"
+	"github.com/justtrackio/gosoline/pkg/log"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type OtelSettings struct {
+	Exporter      string  `cfg:"exporter"`
+	SamplingRatio float64 `cfg:"sampling_ratio" default:"0.05"`
+	SpanLimits
+}
+
+type SpanLimits struct {
+	AttributeValueLengthLimit   int `cfg:"attribute_value_length_limit" default:"-1"`
+	AttributeCountLimit         int `cfg:"attribute_count_limit" default:"128"`
+	EventCountLimit             int `cfg:"event_count_limit" default:"128"`
+	LinkCountLimit              int `cfg:"link_count_limit" default:"128"`
+	AttributePerEventCountLimit int `cfg:"attribute_per_event_count_limit" default:"128"`
+	AttributePerLinkCountLimit  int `cfg:"attribute_per_link_count_limit" default:"128"`
+}
+
+type otelTraceProviderKey struct{}
+
+func ProvideOtelTraceProvider(ctx context.Context, config cfg.Config, logger log.Logger) (trace.TracerProvider, error) {
+	return appctx.Provide(ctx, otelTraceProviderKey{}, func() (trace.TracerProvider, error) {
+		return newOtelTraceProvider(ctx, config, logger)
+	})
+}
+
+func newOtelTraceProvider(ctx context.Context, config cfg.Config, logger log.Logger) (trace.TracerProvider, error) {
+	appId := cfg.AppId{}
+	appId.PadFromConfig(config)
+
+	settings := &OtelSettings{}
+	config.UnmarshalKey("tracing.otel", settings)
+
+	otelExporterFactory := TraceExporters[settings.Exporter]
```

Comment:
You should check that the exporter is actually a valid name and throw a proper error instead of crashing once you call the factory in the next code line